ruby.asciidoc

```diff
@@ -0,0 +1,565 @@
+method on each of the numeric values to get strings that we can
+concatenate to build the output.
+
+To run the program, execute these commands in the measurements
+directory:
+
+[source,shell]
+----
+$ ponyc main.pony
+$ ./measurements
+----
+
+[[getting-functional]]
+Getting functional
+~~~~~~~~~~~~~~~~~~
+
+In this example we just want to use a pure function to return the
+comparison of two integers as a human readable string.
+
+[source,ruby]
+----
+cat functional/functional.rb
+
+def compare(a, b)
+  if (a > b)
+    "a is bigger than b"
+  elsif (a < b)
+    "b is bigger than a"
+  else
+    "a and b are the same"
+  end
+end
+
+puts compare(1, 2)
+----
+
+In the Ruby version, we use a method to avoid needing an instance.
+
+Each comparison and corresponding return is computed with an imperative
+if and return
+
+To run the program, execute this command in the functional directory:
+
+[source,shell]
+----
+$ ruby functional.rb
+----
+
+[source,ponylang]
+----
+$ cat functional/main.pony
+
+actor Main
+  new create(env : Env) =>
+    env.out.print(compare(2,1))
+
+  fun compare(a: I64, b: I64): String =>
+    match (a, b)
+    | if a > b => "a is bigger than b"
+    | if a < b => "b is bigger than a"
+    else "a and b are the same"
+  end
+----
+
+In the Pony version we use a single match expression to yield one result
+based on matching a specific case or using the default else case. Guard
+conditions are used to narrow down the matches. This use of match is
+only a little less verbose than the Ruby version, but it does provide a
+hint of how powerful and concise Pony match expressions can be.
+
+To run the program, execute these commands in the functional directory:
+
+[source,shell]
+----
+$ ponyc main.pony
+$ ./functional
+----
+
+[[testing]]
+Testing
+~~~~~~~
+
+Rubyists love testing, so before we go any farther, let’s talk about
+testing. In Pony, there is a unit testing framework built in called
+`PonyTest`, and it’s pretty simple. Let’s write some very simple code
+and tests to exercise it.
+
+[source,ruby]
+----
+require 'minitest/autorun'
+
+class TestAdd < MiniTest::Unit::TestCase
+  def test_addition
+    assert_equal 4, 2 + 2
+  end
+end
+
+----
+
+Now let's look at the equivalent in Pony:
+
+[source,ponylang]
+----
+$ cat testing/main.pony
+
+use "ponytest"
+
+actor Main is TestList
+  new create(env: Env) =>
+    PonyTest(env, this)
+
+  new make() => None
+
+  fun tag tests(test: PonyTest) => test(_TestAdd)
+
+class _TestAdd is UnitTest
+  fun name():String => "addition"
+
+  fun apply(h: TestHelper) => h.assert_eq[U32](4, 2 + 2)
+
+----
+
+You can compile these tests as you would any other Pony package:
+
+[source,shell]
+----
+$ ponyc .
+----
+
+And you can run the generated binary, which will give you some output:
+
+[source,shell]
+----
+$ ./testing
+1 test started, 0 complete: addition started
+1 test started, 1 complete: addition complete
+---- Passed: addition
+----
+---- 1 test ran.
+---- Passed: 1
+----
```

Comment:
It looks like the formatting got confused here by the `----` in the test suite output.  You may have to adjust this to get the renderer to format it correctly.